packages/preact/test/hooks/useSwatches.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { useSwatches } from "@preact/hooks/useSwatches"
+import { renderHook } from "@testing-library/preact"
+import { describe, expect, it } from "vitest"
+
+const testSKUs = [
+  {
+    id: "SKU-001",
+    customFields: [
```

Comment:
use a helper function here that turns a key value object into an array of key value pairs